packages/styling/src/styles/getFadedOverflowStyle.ts

```diff
@@ -0,0 +1,114 @@
+ * - Apply it to the :after selector for all combination of states the parent of content might have (normal, hover, selected, focus).
+ * - Requires the target to have position set to relative and overflow set to hidden.
+ *
+ * @example
+ * ```tsx
+ * // Assuming the following DOM structure and the different background colors coming from the parent holding the content.
+ * <div className={classNames.parent}>
+ *   <span className={classNames.content}>Overflown Content</span>
+ * </div>
+ * ```
+ * ```ts
+ * // This is how the style set would look in Component.styles.ts
+ * const {bodyBackground, hoverBackground} = theme.semanticColors;
+ *
+ * const styles = {
+ *   parent: [
+ *     backgroundColor: bodyBackground,
+ *     selectors: {
+ *       '&:hover: {
+ *         backgroundColor: hoverBackground
+ *       },
+ *       '$content:after': {
+ *         ...getFadedOverflowStyle(theme, bodyBackground)
+ *       },
+ *       '&:hover $content:after': {
+ *         ...getFadedOverflowStyle(theme, hoverBackground)
+ *       }
+ *     }
+ *   ],
+ *   content: [
+ *     width: '100%',
+ *     display: 'inline-block',
+ *     position: 'relative',
+ *     overflow: 'hidden'
+ *   ]
+ * }
+ * ```
+ * @param theme - The theme object to use.
+ * @param color - The background color to fade out to. Defaults to theme.semanticColors.bodyBackground.
+ * @param direction - The direction of the overflow. Defaults to horizontal.
+ * @param width - The width of the fading overflow. Vertical direction defaults it to 100% vs 20px when horizontal.
+ * @param height - The Height of the fading overflow. Vertical direction defaults it to 50% vs 100% when horizontal.
+ * @returns The style object.
+ */
+export function getFadedOverflowStyle(
+  theme: ITheme,
+  color: string = theme.semanticColors ? theme.semanticColors.bodyBackground : DEFAULT_COLOR,
+  direction: 'horizontal' | 'vertical' = 'horizontal',
+  width: string | number = getDefaultValue('width', direction),
+  height: string | number = getDefaultValue('height', direction)
+): IRawStyle {
+  const rgbColor: IRGB = hex2rgb(color);
+  const rgba = `rgba(${rgbColor.r}, ${rgbColor.g}, ${rgbColor.b}, 0)`; // opacity = 0 for starting color of gradient.
+  const gradientDirection = direction === 'vertical' ? 'to bottom' : 'to right'; // mergeStyles take care of RTL direction.
+
+  return {
+    content: '""',
+    position: 'absolute',
+    right: 0,
+    bottom: 0,
+    width: width,
+    height: height,
+    pointerEvents: 'none',
+    backgroundImage: `linear-gradient(${gradientDirection}, ${rgba} 0%, ${color} 100%)`,
+    selectors: {
+      [HighContrastSelector]: {
+        backgroundImage: 'none'
+      }
+    }
+  };
+}
+
+/**
+ * Helper function to convert a string hex color to an RGB object.
+ *
+ * @param color - Color to be converted from hex to rgba.
+ */
+function hex2rgb(color: string): IRGB {
```

Comment:
all of these should sit in or next to colors.ts
note that colors.ts is quite large since it pulls in all the named colors... so we might want to make a separate minimal color utilities file that doesn't include that

Reply:
@phkuo colors.ts is a part of office-ui-fabric-react package which is not a dependency of styling package where my function resides. This was the main reason I've added this helper function. Should we move the entire colors folder from OUFR to Styling and leave just the index importing from it to prevent a breaking change?